src/coreclr/interpreter/compiler.cpp

```diff
@@ -52,6 +52,12 @@ void DECLSPEC_NORETURN Interp_NOMEM()
     throw std::bad_alloc();
 }
 
+void DECLSPEC_NORETURN Interp_CompilationFail(const char *msg)
+{
+    // This is a fatal error, so we don't return.
```

Comment:
Should this turn into returning `CORJIT_INTERNALERROR` that turns into `COMPlusThrow(kInvalidProgramException, (UINT) IDS_EE_JIT_COMPILER_ERROR)` like in a regular JIT?